astropy/io/fits/tests/test_header.py

```diff
@@ -24,6 +25,23 @@
 from ..util import encode_ascii
 
 
+def test_init_with_dict():
+    dict1 = {'a': 11, 'b': 12, 'c': 13, 'd': 14, 'e': 15}
+    h1 = fits.Header(dict1)
+    for i in dict1:
+        assert i in h1
+        assert dict1[i] == h1[i]
+
+
+def test_init_with_ordereddict():
+    dict1 = collections.OrderedDict({'a': 11, 'b': 12, 'c': 13,
+                                     'd': 14, 'e': 15})
+    h1 = fits.Header(dict1)
+    for i in dict1:
+        assert i in h1
+        assert dict1[i] == h1[i]
```

Comment:
Would be good to check that the order is preserved in this case. (Which means you need to create the OrderedDict with a list of tuples to be sure the order is correct). Also `assert i in h1` seems useless since it will fail in the next line if the key is missing.